### PR TITLE
Adding securityContext options at the pod and container level.

### PR DESCRIPTION
**Pull request by swade1987**
Allowing the configuration of `SecurityContext` at the pod and individual container levels.

### CODE REVIEW

**Comment by saada** [comment]
The tests seem to be failing.
`# Error: Error: Reading chart 'helm-operator' default values (# Default values for helm-operator`.
Not sure why...

**Comment by saada** [comment]
`# ): error converting YAML to JSON: yaml: line 243: did not find expected key`

**Comment by saada** [comment]
```yaml
containerSecurityContext: {}
  helmOperator: {}
  tiller: {}
```

needs to be change to
```yaml
containerSecurityContext:
  helmOperator: {}
  tiller: {}
```

File: chart/helm-operator/templates/deployment.yaml [hunk 1]
```diff
@@ -47,6 +47,10 @@ spec:
       imagePullSecrets:
         - name: {{ .Values.image.pullSecret }}
       {{- end }}
+      {{- if .Values.securityContext }}
+      securityContext:
+{{ toYaml .Values.securityContext | indent 12 }}
```
Comment (saada) [inline chart/helm-operator/templates/deployment.yaml]: I think this should be `| indent: 8` ?

File: chart/helm-operator/templates/deployment.yaml [hunk 2]
```diff
@@ -116,6 +120,10 @@ spec:
       - name: flux-helm-operator
         image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
         imagePullPolicy: {{ .Values.image.pullPolicy }}
+        {{- if .Values.containerSecurityContext.helmOperator }}
+        securityContext:
+{{ toYaml .Values.containerSecurityContext.helmOperator | indent 12 }}
```
Comment (saada) [inline chart/helm-operator/templates/deployment.yaml]: I think this should be `| indent: 10` ?

File: chart/helm-operator/templates/deployment.yaml [hunk 3]
```diff
@@ -279,6 +287,10 @@ spec:
         - --probe-listen=:44135
         - --storage={{ .Values.tillerSidecar.storage }}
         imagePullPolicy: {{ .Values.image.pullPolicy }}
+        {{- if .Values.containerSecurityContext.tiller }}
+        securityContext:
+ {{ toYaml .Values.containerSecurityContext.tiller | indent 12 }}
```
Comment (saada) [inline chart/helm-operator/templates/deployment.yaml]: I think this should be `| indent: 10` ?

File: chart/helm-operator/values.yaml [hunk 4]
```diff
@@ -237,3 +237,9 @@ dashboards:
   # If enabled, helm-operator will create a configmap with a dashboard in json that's going to be picked up by grafana
   # See https://github.com/helm/charts/tree/master/stable/grafana#configuration - `sidecar.dashboards.enabled`
   enabled: false
+
+securityContext: {}
+
+containerSecurityContext: {}
```
Comment (saada) [inline chart/helm-operator/values.yaml]: please remove `{}` here

**Review by saada — CHANGES_REQUESTED** [review]

**Comment by saada** [comment]
Thanks for contributing! Just a few pointers to get these tests passing

**Comment by saada** [comment]
@hiddeco can we rerun tests?

**Comment by hiddeco** [comment]
Rebasing and force-pushing will trigger a new build while also getting rid of the `Merge branch 'master' into master` commit.

File: chart/helm-operator/Chart.yaml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 appVersion: "1.2.0"
-version: 1.1.0
+version: 1.1.1
```
Comment (hiddeco) [inline chart/helm-operator/Chart.yaml]: FYI: we do not bump versions on changes, but bump it on release.

**Review by hiddeco — APPROVED** [review]
Thanks @swade1987 :tulip:

**Pull request merged**